conformance/lang/values-types-and-variables/array_type.balt

```diff
@@ -0,0 +1,332 @@
+Test-case: output
+Description: Test array declaration with variable array length
+Label: array-type-descriptor, list-type-descriptor, list-constructor-expr, int-type-descriptor, assignment-stmt,
+       int-literal
+
+public function main() {
+    int[] d = [1, 2];
+    io:println(d); // @output [1,2]
+    d = [1];
+    io:println(d); // @output [1]
+    d = [1, 6555, 4, 449, 9];
+    io:println(d); // @output [1,6555,4,449,9]
+}
+
+Test-case: output
+Description: Test basic array type and value
+Label: array-type-descriptor, list-type-descriptor, int-type-descriptor, int-literal, list-constructor-expr
+
+public function main() {
+    int[5] a = [1, 2, 3, 4, 5];
+    io:println(a); // @output [1,2,3,4,5]
+
+    int[4] b = [20, 65];
+    io:println(b); // @output [20,65,0,0]
+}
+
+Test-case: error
+Description: Test error in assigning a value that does not belong to shapes denoted by the array type descriptor
+Label: array-type-descriptor, list-type-descriptor, int-type-descriptor, list-constructor-expr, int-literal
+
+public function main() {
+    int[2] a = [1, 2, 3, 4, 5]; // @error length of array value is greater than 2
+}
+
+Test-case: output
+Description: Test array declaration with array length as constant-reference-expr
+Label: array-type-descriptor, list-type-descriptor, int-literal, list-constructor-expr, module-const-decl,
+       constant-reference-expr
+
+const LENGTH = 2;
+
+public function main() {
+    int[LENGTH] a = [1, 255];
+    io:println(a); // @output [1,255]
+}
+
+Test-case: error
+Description: Test error in array declaration with array length as negative constant-reference-expr
+Fail-Issue: ballerina-lang#32269
+Label: array-type-descriptor, list-type-descriptor, int-literal, list-constructor-expr, module-const-decl,
+       constant-reference-expr
+
+const LENGTH2 = -1;
+
+public function main() {
+    int[LENGTH2] bool; // @error constant-reference-expr must evaluate to non-negative integer
+}
+
+Test-case: error
+Description: Test invalid array lengths
+Label: array-type-descriptor, list-type-descriptor, int-type-descriptor, list-constructor-expr,
+       string-literal, int-literal, float-literal, boolean-literal
+
+public function main() {
+    int length = 3;
+    int[length] a = [1, 2, 3]; // @error `length` must be a const-ref-expr
+
+    int["c"] b = [1]; // @error invalid array length
+
+    int[5.4] c = [4, 5]; // @error array length must be non-negative integer
+
+    int[true] d = [4, 5]; // @error invalid array length
+}
+
+Test-case: error
+Description: Test array length as negative integer
+Fail-Issue: ballerina-lang#32394
+Label: array-type-descriptor, list-type-descriptor, int-type-descriptor, list-constructor-expr, int-literal
```

Comment:
Since we are using a negative integer, I think, here we need to add the 'unary-expr' label also right?

Reply:
Yes, will update